### PR TITLE
Stop reload on network change

### LINKED ISSUE #3599: Stop reloading dapps on network change
Expected behavior: Chain change in MM UI should not trigger a whole refresh of the browser page.

I experience this in Chrome AND firefox, with my react application.

Any input on this ?

Many thanks

### PR DESCRIPTION
**Pull request by rekmarks**
Per issue #3599, the goal is to stop reloading dapps on network change. The responsible code was in `auto-reload.js`. It contained some other functionality. To resolve the problem, the following steps were taken:

- moved global `web3` export and deprecation warning functionality from `auto-reload.js` to `inpage.js`
- removed `auto-reload.js` import in `inpage.js`
- removed `auto-reload.js`

*Note:* In the commit `wip`, I had only removed the reload functionality from `auto-reload.js`.

Functionality verified locally with all tests passing. Resolves #3599.

**Edit:** This feature was not included in 4.8.0 to allow dapp developers to respond to the change. Expected deployment now in the beginning of July.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -118,7 +118,6 @@
     "fast-levenshtein": "^2.0.6",
     "file-loader": "^1.1.11",
     "fuse.js": "^3.2.0",
-    "gulp": "github:gulpjs/gulp#4.0",
```
Comment (danfinlay) [inline package.json]: Is this a safe change? Does a fresh install still let you build, if you don't have `gulp` globally installed?

File: app/scripts/inpage.js [hunk 2]
```diff
@@ -38,8 +37,24 @@ web3.setProvider = function () {
   log.debug('MetaMask - overrode web3.setProvider')
 }
 log.debug('MetaMask - injected web3')
-// export global web3, with usage-detection
-setupDappAutoReload(web3, inpageProvider.publicConfigStore)
+
+// export global web3, with usage-detection and deprecation warning
+let hasBeenWarned = false
+global.web3 = new Proxy(web3, {
```
Comment (danfinlay) [inline app/scripts/inpage.js]: Reasonable place to move it, I assume you've manually verified the warning still works?

**Review by danfinlay — DISMISSED** [review]
Could use a changelog entry. If you've verified it works, then it looks good to me.

File: app/scripts/inpage.js (hunk 2 above)
Comment (rekmarks) [reply app/scripts/inpage.js]: Yes, manually verified in Chrome.

**Comment by rekmarks** [comment]
Added commits to address review comments.
- updated `CHANGELOG.md`
- undid changes to `package.json` and `package-lock.json`

**Comment by Platinumwrist** [comment]
Ok

**Review by danfinlay — APPROVED** [review]
LGTM

**Comment by stefek99** [comment]
Thanks!

Warning about depreciation in `auto-reload` was pretty random :)

**Comment by rekmarks** [comment]
@stefek99 our pleasure.

And yeah, we thought so too 😅

**Comment by safareli** [comment]
when release with this change is expected to be made?

**Comment by lazaridiscom** [comment]
@safareli , it looks like this is already within version 4.8.0

[link](https://github.com/MetaMask/metamask-extension/pull/4603/commits/73a9aaa8338c2d8a2960a79f02369caaa21b3c95)

**Comment by safareli** [comment]
Strange, I created an issue #4673 as apps are still reloaded on 4.8.0

**Comment by rekmarks** [comment]
@safareli We delayed the release of this feature in order to allow dapp developers to respond to the change. We expect to deploy it in the next week or two.

**Comment by lazaridiscom** [comment]
> Strange, I created an issue #4673 as apps are still reloaded on 4.8.0

@safareli, the change itself was reverted  within https://github.com/MetaMask/metamask-extension/commit/3a6cc3c8fd1cf16b744af61104472887d6d37fb3, but the changelog was not reverted for some reason.

**Comment by safareli** [comment]
Thanks for clarification

**Comment by W3stside** [comment]
Hi guys - what is the anticipated release for this? I know it says a week or two but thought I'd check in, cheers!

**Comment by danfinlay** [comment]
We're going to take longer to roll this out. We have another breaking change we're going to roll out soon, so we are going to publish a second post that includes that one ([not injecting web3 by default](https://github.com/MetaMask/metamask-extension/pull/4703)), along with strong "test your dapp" tools (like a special build of MetaMask), and some active campaigning to ensure the most popular dapps are compatible before we switch over.

**Comment by lazaridiscom** [comment]
@danfinlay , although this is not always technically applicable, one possible clean way of introducing a breaking change would be this one:

* keep the original behavior, if possible 100% non-altered
* Implement the breaking behavior, without altering the original behaviour
    * try to use clean/strict refactoring
    * if unavoidable: use code-duplication (in order to keep original behavior unaltered)
* Provide an option which activates the breaking behavior
* The original behavior is active by default (new/breaking behavior is opt-in)
* Provide a deprecation warning, announcing when the original behavior will be dropped
* drop the original behavior once the time comes.

A clean code-base as starting-point makes things of course easier, thus -under circumstances- it could benefit to do a general refactoring/redesign prior to implementing the breaking change.

(Note: I do not know the relevant code-base, so possibly all the above do not apply to the issue at hand)

**Pull request merged**